photutils/psf/psfphotometry.py

```diff
@@ -0,0 +1,282 @@
+        """
+
+        result_tab = Table([[], [], [], [], []],
+                           names=('id', 'group_id', 'x_fit', 'y_fit',
+                                  'flux_fit'),
+                           dtype=('i4', 'i4', 'f8', 'f8', 'f8'))
+
+        star_groups = star_groups.group_by('group_id')
+        
+        groups_order = []
+        for g in star_groups.groups:
+            groups_order.append(len(g))
+
+        N = len(groups_order)
+        while N > 0:
+            curr_order = np.min(groups_order)
+            n = 0
+            N = len(groups_order)
+            while(n < N):
+                if curr_order == len(star_groups.groups[n]):
+                    group_psf = self._get_sum_psf_model(star_groups.groups[n])
+                    x, y, data = self._get_shape_and_data(shape=self.fitshape,\
+                                             star_group=star_groups.groups[n],
+                                                          image=image)
+                    fit_model = self.fitter(group_psf, x, y, data)
+                    print(fit_model)
+                    param_table = self._model_params2table(fit_model,\
+                                                        star_groups.groups[n])
+                    result_tab = vstack([result_tab, param_table])
+                    image = subtract_psf(image, self.psf, param_table)
+                    N = N - 1
+                n += 1
+        return result_tab, image
+
+    def _model_params2table(self, fit_model, star_group):
+        """
+        Place fitted parameters into an astropy table.
+        
+        Parameters
+        ----------
+        fit_model : Fittable2DModel
+        star_group : ~astropy.table.Table
+        
+        Returns
+        -------
+        param_tab : ~astropy.table.Table
+            Table that contains the fitted parameters.
+        """
+
+        param_tab = Table([[], [], [], [], []],
+                          names=('id', 'group_id', 'x_fit', 'y_fit',
+                                 'flux_fit'),
+                          dtype=('i4', 'i4', 'f8', 'f8', 'f8'))
+        if np.size(fit_model) == 1:
+            param_tab.add_row([[star_group['id'][0]],
+                               [star_group['group_id'][0]],
+                               [getattr(fit_model,'x_0').value],
+                               [getattr(fit_model, 'y_0').value],
+                               [getattr(fit_model, 'flux').value]])
+        else:
+            for i in range(np.size(fit_model)):
+                param_tab.add_row([[star_group['id'][i]],
+                                   [star_group['group_id'][i]],
+                                   [getattr(fit_model,'x_0_'+str(i)).value],
+                                   [getattr(fit_model, 'y_0_'+str(i)).value],
+                                   [getattr(fit_model, 'flux_'+str(i)).value]])
+        return param_tab
+
+    def _get_shape_and_data(self, shape, star_group, image):
+        """
```

Comment:
This is very similar to 
https://astropy.readthedocs.io/en/latest/api/astropy.nddata.utils.extract_array.html#astropy.nddata.utils.extract_array
(see also the list of related functions in https://astropy.readthedocs.io/en/latest/nddata/index.html#module-astropy.nddata.utils).
In fact I wrote `extract_array` specifically to do PSF photometry. It was the longest PR I ever had with about 100 comments (see https://github.com/astropy/astropy/pull/3333 - and you thought we are thorough on your pull requests ;-)  ).
Please don't make that wasted by implementig it again here. Also (based on the experiences in astropy #3333) I can tell you that the implementation here will have trouble on the edges, and will not allow me to extract sources whose center is not on the chip (but if they are bright, we can still fit a PSF to the wings and get a reasonable flux, and I have a dataset where I want to do just that).